cpp/CMakeLists.txt

```diff
@@ -342,11 +342,13 @@ add_library(cugraph SHARED
     src/traversal/sssp.cu
     src/link_prediction/jaccard.cu
     src/link_prediction/overlap.cu
-    src/converters/nvgraph.cu
     src/converters/renumber.cu
     src/converters/COOtoCSR.cu
-    src/community/nvgraph_gdf.cu
+    src/community/nvgraph_clustering.cu
+    src/community/louvain.cu
     src/community/ECG.cu
+    src/community/triangles_counting.cu
```

Comment:
I noticed some files were moved out of `nvgraph` directory but others (most of the spectral clustering backend and some helpers) are still there. What are the selection criteria? Should we rename or remove the directory?

Reply:
I made several attempts at doing this and settled on this one.  Here's my criteria:

1. The primary entry points were created in the src/community directory (where nvgraph_gdf.cu was)
1. The top level of implement was moved into the src/community directory co-resident with the primary entry point
1. All library-like things and headers were left under nvgraph.

In one of my attempts, I actually started refactoring a bunch of the other components more completely, but I realized two things.  First, it would be a great deal of programming effort.  Second, the new API that we are developing as part of OPG (and the GraphBLAS work) would and up replacing much of this.  What I settled on as a reasonable compromise was to completely refactor things that were in src/community (or that I moved to src/community) and do selective editing of the nvgraph core functionality to make it work with the new graph structure.

We can certainly go further if we like, although I'd be inclined to do that in a separate PR rather than make this even bigger.